### PR TITLE
Migrate DKC3 to APDeltaPatch

### PR DESCRIPTION
**Pull request by PoryGone**
In support of #835 to solve DKC3 patches not showing on Room Page.

### CODE REVIEW

**Review by Ijwu — APPROVED** [review]
Approved, but I did leave a comment I'd like an answer to.

File: worlds/dkc3/__init__.py [hunk 1]
```diff
@@ -145,12 +145,16 @@ def generate_output(self, output_directory: str):
 
             rompath = os.path.join(output_directory, f'AP_{world.seed_name}{outfilepname}.sfc')
             rom.write_to_file(rompath)
-            Patch.create_patch_file(rompath, player=player, player_name=world.player_name[player], game=Patch.GAME_DKC3)
-            os.unlink(rompath)
             self.rom_name = rom.name
         except:
```
[the remark this answers is not in the document]
Comment (PoryGone) [reply worlds/dkc3/__init__.py:149]: It's a classic case of "I followed the way another world did it", in this case: Super Metroid. The structure here was not a deliberate choice on my part, but just simulating another example that already worked.

[the remark this answers is not in the document]
Comment (Ijwu) [reply worlds/dkc3/__init__.py:149]: Well, let's make it better. I'm going to assume you know what all of try-catch-else-finally do, so I'll leave it to you to refactor to something more fitting.

[the remark this answers is not in the document]
Comment (PoryGone) [reply worlds/dkc3/__init__.py]: I have adjusted the logic here.

**Pull request merged**